internal/controller/streams.go

```diff
@@ -240,7 +243,7 @@ func (sm *streamManager) buildResourceApiGroupMap(resources []string, clientset
 }
 
 // StreamResources handles the resource stream.
-func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger, cancel context.CancelFunc) error {
+func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger, cancel context.CancelFunc, snapshotCommitSignal chan struct{}) error {
```

Comment:
How about passing a `func()` instead of directly the `chan`, to make this a bit more abstract?
Like the `CancelFunc` of a cancelable `Context`...
```suggestion
func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger, cancel context.CancelFunc, signalSnapshotCommit func()) error {

Reply:
Sure that is fine with me!

Reply:
I tried this and it did not make sense since chans make more sense for signaling.